tuf/ngclient/_internal/requests_fetcher.py

```diff
@@ -43,7 +43,7 @@ def __init__(self) -> None:
         # improve efficiency, but avoiding sharing state between different
         # hosts-scheme combinations to minimize subtle security issues.
         # Some cookies may not be HTTP-safe.
-        self._sessions: Dict[str, requests.Session] = {}
+        self._sessions: Dict[Tuple[str, Optional[str]], requests.Session] = {}
```

Comment:
nitpick: no need for Optional: hostname is guaranteed to be a string (it may be an empty string but it's not None).